pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -762,29 +763,34 @@ private void handleSubquery(Expression expression, long requestId, JsonNode json
    * - Case-insensitive cluster
    * - Table name in the format of [database_name].[table_name]
    *
-   * Drop the database part if there is no existing table in the format of [database_name].[table_name], but only
-   * [table_name].
+   * If the PinotConfiguration does not allow dots in table name, will do the below:
+   * Drop the prefix part for the format of [database_name].[table_name], keep only [table_name].
+   * If the PinotConfiguration allows dots in table name, we will not do the split
+   * @param tableName the table name in the query
+   * @param tableCache the table case-sensitive cache
+   * @param routingManager the routing mananger for testing whether a route exists
+   * @return the table name in the format of [database_name].[table_name]
    */
-  private String getActualTableName(String tableName) {
+  @VisibleForTesting
+  static String getActualTableName(String tableName, TableCache tableCache, BrokerRoutingManager routingManager) {
     // Use TableCache to handle case-insensitive table name
-    if (_tableCache.isCaseInsensitive()) {
-      String actualTableName = _tableCache.getActualTableName(tableName);
+    if (tableCache.isIgnoreCase()) {
```

Comment:
I just realize that we have enhanced `tableCache` in #8475 to store table name regardless of the case sensitivity. We can actually significantly simplify this method by always using `tableCache` to resolve the actual table name:
```suggestion
static String getActualTableName(String tableName, TableCache tableCache) {
    String actualTableName = tableCache.getActualTableName(tableName);
    if (actualTableName != null) {
      return actualTableName;
    }
    
    // Check if table is in the format of [database_name].[table_name]
    String[] tableNameSplits = StringUtils.split(tableName, ".", 2);
    if (tableNameSplits.length == 2) {
      actualTableName = tableCache.getActualTableName(tableNameSplits[1]);
      if (actualTableName != null) {
        return actualTableName;
      }
    }

    return tableName;
  }
```